static/js/input.js

```diff
@@ -442,6 +447,41 @@ function guessColumnType(arr) {
 }
 
 
+/**
+ * Calculates the entropy of an array
+ * @function calcEntropy
+ * @param {Array} arr - array of values
+ * @returns {Number} entropy - entropy of array
+ */
+function calcEntropy(arr, match) {
+  const unique = [...new Set(arr)];
+  const base = unique.length;
+  let probs = [];
+
+  // calculates probability of occurence of search value
+  function calcProb(match) {
+    const n = arr.length;
+
+    let count = 0;
+    for (let i = 0; i < n; i++) {
+      if (arr[i] === match) {
+        count++;
+      }
+    }
+
+    return count / n;
+  }
+
+  for (let val of unique) probs.push(calcProb(val));
```

Comment:
Here, you are iterating the array once for each unique value. If there are many unique values, this will be slow. A more optimal way is to iterate the array only once and calculate frequencies for all unique values. An example is the `listCats` function which I already [implemented](https://github.com/qiyunlab/binarena/blob/master/static/js/util.js#L356) In your case, you don't need the values but only want the frequencies, and you don't need to skip false values, so you can modify it accordingly.